WordPress/gutenberg
# Writing flow: Backspace at beginning of first paragraph block prevents block from being deleted

Fixes #56303
Related to #55134

## What?
This PR fixes an issue where pressing the backspace key at the beginning of a paragraph block with content causes the block to be deleted.

### Before 

https://github.com/WordPress/gutenberg/assets/54422211/7f0ccf27-d486-4e60-9437-8462be6a1d63

### After

https://github.com/WordPress/gutenberg/assets/54422211/bdae7725-f403-4440-b98f-e61db4606b3f

## Why?

I think the current implementation deletes a block when you press the backspace key at the beginning of the first block. This is a problem if the paragraph or heading already has content.

## How?
I updated the logic to have the following behavior when the BACKSPACE key is pressed in the first block. I think this is expected behavior.

- If it is the default block, i.e. a paragraph block, do nothing
- If it is not a default block, convert it to the default block. This means that headings are converted to paragraph blocks while preserving their content.

## Testing Instructions

- Insert a Paragraph block at the beginning of the content and enter the text.
- Press the backspace at the beginning of the Paragraph block.
  - trunk: The block will be removed and you should now have an empty paragraph.
  - This PR: Nothing should happen.
- Insert a heading block at the beginning of the content and enter the text.
- Press the backspace key at the beginning of the Heading block.
  - trunk: The block will be removed and you should now have an empty paragraph.
  - This PR: Should be transformed to a paragraph block.
